src/main/java/eod/event/relay/EventReceiver.java

```diff
@@ -1,12 +1,11 @@
-package eod.event.listener;
+package eod.event.relay;
 
 import eod.GameObject;
-import eod.Player;
 import eod.event.Event;
 
 import java.util.ArrayList;
 
-public interface EventListener {
+public interface EventReceiver extends GameObject {
```

Comment:
why extends GameObject?

Reply:
since a receiver is always a thing that needs to teardown